src/uinputevents.h

```diff
@@ -0,0 +1,51 @@
+// This file is part of Projecteur - https://github.com/jahnf/projecteur - See LICENSE.md and README.md
+// Virtal Device to emit customized events from Projecteur device
+// The spotlight.cc grabs mouse inputs from Logitech Spotlight device.
+// This module is used when the input events are supposed to be forwarded to the system.
+# pragma once
+
+#include <iostream>
+#include <memory>
+#include <linux/uinput.h>
+#include <unistd.h>
+
+#include <QDebug>
+
+using namespace std;
+
+class uinputEvents;
+
+class uinputEvents{
+  private:
+    // Device that can act as virtual keyboard and mouse
+    struct uinput_user_dev uinp;
+	static int uinp_fd;
+
+    uinputEvents(){
+      uinp_fd = -1;
+      // Try to setup the device. If it fails exit.
+      if (setup_uinputDevice() != 1)
+         exit(1);
+    }
+
+    int setup_uinputDevice();
+
+
+  public:
+	uinputEvents(uinputEvents const&) = delete;
+	void operator=(uinputEvents const&) = delete;
+	~uinputEvents() {
+	  ioctl(uinp_fd, UI_DEV_DESTROY);
+	  close(uinp_fd);
+	  qDebug("uinput Device Closed");
+	}
+
+    static shared_ptr<uinputEvents> getInstance() {
```

Comment:
Generally I disagree to make this class a singleton. I don't want to use singletons - only if I absolutely have to. Same as with global variables.

Suggestion: 
* `UInputEvents` class has a constructor that takes a name as argument (uinput device name)
* Spotlight holds a UInputEvents instance as a member.

Reply:
The second suggestion 'Spotlight holds a UInputEvents instance as a member.' is already implemented.

Reply:
Well, kind of, but not really and it is not very clean:
You have a raw pointer to a shared_ptr data returned from a singleton. Alone this description should let you shudder :) And there you should have a shared_ptr<uinputevents> as a member in spotlight.

Anyway, since a singleton is not necessary here please implement UInputEvents as a non-singleton.